### PR TITLE
Added modification to ProGuard setting

### PR DESCRIPTION
**Pull request by lwierzbicki**
This PR covers issue #805 .

### CODE REVIEW

**Comment by lwierzbicki** [comment]
Additionally last commit cover issue #1632

**Comment by lwierzbicki** [comment]
Last commit covers issue #1628 .

File: Document/0x04f-Testing-Network-Communication.md [hunk 1]
```diff
@@ -165,7 +169,7 @@ The following configuration files need to be changed and adjusted accordingly:
 - hostapd.conf
 
     ```
-    # Name of the WiFi interface we use 
+    # Name of the WiFi interface we use
```
Comment (cpholguera) [inline Document/0x04f-Testing-Network-Communication.md]:
```suggestion
    # Name of the WiFi interface we use
```

File: Document/0x04f-Testing-Network-Communication.md [hunk 2]
```diff
@@ -221,14 +225,14 @@ To be able to get a man-in-the-middle position you need to run the above configu
     $ wpa_supplicant -B -i wlan0 -c wpa_supplicant.conf
     # run DNS server
     $ dnsmasq -C dnsmasq.conf -d
-    # enable routing 
+    # enable routing
```
Comment (cpholguera) [inline Document/0x04f-Testing-Network-Communication.md]:
```suggestion
    # enable routing
```

File: Document/0x04f-Testing-Network-Communication.md [hunk 3]
```diff
@@ -221,14 +225,14 @@ To be able to get a man-in-the-middle position you need to run the above configu
...
     # run DNS server
     $ dnsmasq -C dnsmasq.conf -d
-    # enable routing 
+    # enable routing  
     $ echo 1 > /proc/sys/net/ipv4/ip_forward
     # iptables will NAT connections from AP network interface to the target network interface
     $ iptables --flush
     $ iptables --table nat --append POSTROUTING --out-interface wlan0 -j MASQUERADE
-    $ iptables --append FORWARD --in-interface wlan1 -j ACCEPT 
+    $ iptables --append FORWARD --in-interface wlan1 -j ACCEPT
```
Comment (cpholguera) [inline Document/0x04f-Testing-Network-Communication.md]:
```suggestion
    $ iptables --append FORWARD --in-interface wlan1 -j ACCEPT
```

File: Document/0x06f-Testing-Local-Authentication.md [hunk 4]
```diff
@@ -242,13 +242,7 @@ Unlike macOS and Android, iOS currently (at iOS 12) does not support temporarine
 
 #### OWASP Mobile Top 10 2016
```
Comment (cpholguera) [inline Document/0x06f-Testing-Local-Authentication.md]:
```suggestion
```

File: Document/0x06f-Testing-Local-Authentication.md [hunk 5]
```diff
@@ -242,13 +242,7 @@ Unlike macOS and Android, iOS currently (at iOS 12) does not support temporarine
...
```
Comment (cpholguera) [inline Document/0x06f-Testing-Local-Authentication.md]:
```suggestion
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 6]
```diff
@@ -502,21 +502,21 @@ We will use the vulnerable password manager app [Sieve](https://github.com/mwrla
 Identify all defined `<provider>` elements:
 
 ```xml
-<provider 
-      android:authorities="com.mwr.example.sieve.DBContentProvider" 
-      android:exported="true" 
-      android:multiprocess="true" 
+<provider
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
<provider
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 7]
```diff
@@ -502,21 +502,21 @@ We will use the vulnerable password manager app [Sieve](https://github.com/mwrla
 Identify all defined `<provider>` elements:
 
 ```xml
-<provider 
-      android:authorities="com.mwr.example.sieve.DBContentProvider" 
-      android:exported="true" 
-      android:multiprocess="true" 
+<provider  
+      android:authorities="com.mwr.example.sieve.DBContentProvider"
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
      android:authorities="com.mwr.example.sieve.DBContentProvider"
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 8]
```diff
@@ -502,21 +502,21 @@ We will use the vulnerable password manager app [Sieve](https://github.com/mwrla
 Identify all defined `<provider>` elements:
 
 ```xml
-<provider 
-      android:authorities="com.mwr.example.sieve.DBContentProvider" 
-      android:exported="true" 
-      android:multiprocess="true" 
+<provider  
+      android:authorities="com.mwr.example.sieve.DBContentProvider"  
+      android:exported="true"
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
      android:exported="true"
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 9]
```diff
@@ -502,21 +502,21 @@ We will use the vulnerable password manager app [Sieve](https://github.com/mwrla
...
 
 ```xml
-<provider 
-      android:authorities="com.mwr.example.sieve.DBContentProvider" 
-      android:exported="true" 
-      android:multiprocess="true" 
+<provider  
+      android:authorities="com.mwr.example.sieve.DBContentProvider"  
+      android:exported="true"  
+      android:multiprocess="true"
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
      android:multiprocess="true"
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 10]
```diff
@@ -502,21 +502,21 @@ We will use the vulnerable password manager app [Sieve](https://github.com/mwrla
...
-      android:exported="true" 
-      android:multiprocess="true" 
+<provider  
+      android:authorities="com.mwr.example.sieve.DBContentProvider"  
+      android:exported="true"  
+      android:multiprocess="true"  
       android:name=".DBContentProvider">
-    <path-permission 
-          android:path="/Keys" 
+    <path-permission
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
    <path-permission
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 11]
```diff
@@ -502,21 +502,21 @@ We will use the vulnerable password manager app [Sieve](https://github.com/mwrla
...
-      android:multiprocess="true" 
+<provider  
+      android:authorities="com.mwr.example.sieve.DBContentProvider"  
+      android:exported="true"  
+      android:multiprocess="true"  
       android:name=".DBContentProvider">
-    <path-permission 
-          android:path="/Keys" 
+    <path-permission  
+          android:path="/Keys"
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
          android:path="/Keys"
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 12]
```diff
@@ -502,21 +502,21 @@ We will use the vulnerable password manager app [Sieve](https://github.com/mwrla
...
           android:readPermission="com.mwr.example.sieve.READ_KEYS"
           android:writePermission="com.mwr.example.sieve.WRITE_KEYS"
      />
 </provider>
-<provider 
-      android:authorities="com.mwr.example.sieve.FileBackupProvider" 
-      android:exported="true" 
-      android:multiprocess="true" 
+<provider  
+      android:authorities="com.mwr.example.sieve.FileBackupProvider"
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
      android:authorities="com.mwr.example.sieve.FileBackupProvider"
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 13]
```diff
@@ -502,21 +502,21 @@ We will use the vulnerable password manager app [Sieve](https://github.com/mwrla
...
+          android:path="/Keys"  
           android:readPermission="com.mwr.example.sieve.READ_KEYS"
           android:writePermission="com.mwr.example.sieve.WRITE_KEYS"
      />
 </provider>
-<provider 
-      android:authorities="com.mwr.example.sieve.FileBackupProvider" 
-      android:exported="true" 
-      android:multiprocess="true" 
+<provider
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
<provider
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 14]
```diff
@@ -502,21 +502,21 @@ We will use the vulnerable password manager app [Sieve](https://github.com/mwrla
...
           android:writePermission="com.mwr.example.sieve.WRITE_KEYS"
      />
 </provider>
-<provider 
-      android:authorities="com.mwr.example.sieve.FileBackupProvider" 
-      android:exported="true" 
-      android:multiprocess="true" 
+<provider  
+      android:authorities="com.mwr.example.sieve.FileBackupProvider"  
+      android:exported="true"
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
      android:exported="true"
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 15]
```diff
@@ -502,21 +502,21 @@ We will use the vulnerable password manager app [Sieve](https://github.com/mwrla
...
      />
 </provider>
-<provider 
-      android:authorities="com.mwr.example.sieve.FileBackupProvider" 
-      android:exported="true" 
-      android:multiprocess="true" 
+<provider  
+      android:authorities="com.mwr.example.sieve.FileBackupProvider"  
+      android:exported="true"  
+      android:multiprocess="true"
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
      android:multiprocess="true"
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 16]
```diff
@@ -381,13 +381,13 @@ To determine whether all the `android.util.Log` class' logging functions have be
...
 ```java
-Log.v("Private key [byte format]: " + key);
+Log.v("Private key tag","Private key [byte format]: " + key);
 ```
 
 The compiled bytecode, however, is equivalent to the bytecode of the following log statement, which constructs the string explicitly:
 
 ```java
-Log.v(new StringBuilder("Private key [byte format]: ").append(key.toString()).toString());
+Log.v(new StringBuilder("Private key tag"),new StringBuilder("Private key [byte format]: ").append(key.toString()).toString());
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
Log.v("Private key tag", new StringBuilder("Private key [byte format]: ").append(key.toString()).toString());
```

File: Document/0x05d-Testing-Data-Storage.md [hunk 17]
```diff
@@ -381,13 +381,13 @@ To determine whether all the `android.util.Log` class' logging functions have be
 Note that the example above only ensures that calls to the Log class' methods will be removed. If the string that will be logged is dynamically constructed, the code that constructs the string may remain in the bytecode. For example, the following code issues an implicit `StringBuilder` to construct the log statement:
 
 ```java
-Log.v("Private key [byte format]: " + key);
+Log.v("Private key tag","Private key [byte format]: " + key);
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
Log.v("Private key tag", "Private key [byte format]: " + key);
```

File: Document/0x05d-Testing-Data-Storage.md (hunk 16 above)
Comment (cpholguera) [reply Document/0x05d-Testing-Data-Storage.md]: Why using stringbuilder here and not in the line above? Or vice versa

File: Document/0x05d-Testing-Data-Storage.md [hunk 18]
```diff
@@ -363,7 +363,7 @@ You should check the apps' source code for logging mechanisms by searching for t
 
 While preparing the production release, you can use tools like `ProGuard` (included in Android Studio). [ProGuard](https://www.guardsquare.com/en/products/proguard "ProGuard") is a free Java class file shrinker, optimizer, obfuscator, and preverifier. It detects and removes unused classes, fields, methods, and attributes and can also be used to delete logging-related code.
 
-To determine whether all the `android.util.Log` class' logging functions have been removed, check the ProGuard configuration file (_proguard-project.txt_) for the following options:
+To determine whether all the `android.util.Log` class' logging functions have been removed, check the ProGuard configuration file (_proguard-rules.pro_) for the following options (according to [example of removing logging code](https://www.guardsquare.com/en/products/proguard/manual/examples#logging "ProGuard\'s exmaple of removing logging code") and [enabling ProGuard in Android Studio project ](https://developer.android.com/studio/build/shrink-code#enable "Android Developer - Enable shrinking, obfuscation, and optimization")):
```
Comment (cpholguera) [inline Document/0x05d-Testing-Data-Storage.md]:
```suggestion
To determine whether all logging functions from the `android.util.Log` class have been removed, check the ProGuard configuration file (proguard-rules.pro) for the following options (according to this [example of removing logging code](https://www.guardsquare.com/en/products/proguard/manual/examples#logging "ProGuard\'s exmaple of removing logging code") and this article about [enabling ProGuard in an Android Studio project](https://developer.android.com/studio/build/shrink-code#enable "Android Developer - Enable shrinking, obfuscation, and optimization")):
```

File: Document/0x05d-Testing-Data-Storage.md (hunk 16 above)
Comment (lwierzbicki) [reply Document/0x05d-Testing-Data-Storage.md]: Updating could be tricky. I think probably that there could be some leftover in bytecode, which is explained in that paragraph. Check again - tag is not in decompiled code. However log message is, example:
```
 public void sendMessage(View paramView) {
    Intent intent = new Intent((Context)this, DisplayMessageActivity.class);
    EditText editText = (EditText)findViewById(2131165238);
    String str = editText.getText().toString();
    StringBuilder stringBuilder = new StringBuilder("Sending message:");
    stringBuilder.append(editText.toString());
    stringBuilder.toString();
    intent.putExtra("com.example.myfirstapp.MESSAGE", str);
    startActivity(intent);
  }
```
In that case your fix is perfect :) Thanks for spotting.

File: Document/0x04f-Testing-Network-Communication.md [hunk 19]
```diff
@@ -113,9 +113,9 @@ Following scenarios are possible:
 
 The scenario with an external USB WiFi card require that the card has the capability to create an access point. Additionally, you need to install some tools and/or configure the network to enforce a man-in-the-middle position (see below). You can verify if your WiFi card has AP capabilities by using the command `iwconfig` on Kali Linux:
 
-    ```shell
-    $ iw list | grep AP 
-    ```
+```shell
+    $ iw list | grep AP
```
Comment (cpholguera) [inline Document/0x04f-Testing-Network-Communication.md]:
```suggestion
    $ iw list | grep AP
```

**Pull request merged**